src/BenchmarksDriver/Program.cs

```diff
@@ -141,103 +142,113 @@ public static int Main(string[] args)
                     }
                 }
 
-                Uri clientJobUri = null;
-                try
-                {
-                    Log($"Starting scenario {scenario} on benchmark client...");
+                Log("Warmup");
+                await RunClientJob(scenario, clientUri, serverBenchmarkUri);
 
-                    var clientJobsUri = new Uri(clientUri, "/jobs");
+                Log("Benchmark");
+                var clientJob = await RunClientJob(scenario, clientUri, serverBenchmarkUri);
 
-                    var clientJob = new ClientJob(_clientJobs[scenario]) { ServerBenchmarkUri = serverBenchmarkUri };
-                    var clientContent = JsonConvert.SerializeObject(clientJob);
+                Debug.Assert(clientJob.State == ClientState.Completed);
```

Comment:
Why use Trace.Assert() instead of Debug.Assert()?  AFAIK, the only difference is the former works in both Release and Debug, which doesn't really matter since this is test driver code, but I've always used Debug.Assert() to avoid slowing down Release builds.